docs/examples/custom_channel.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright (c) 2024 ZettaScale Technology
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+# which is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+#
+# Contributors:
+#   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+#
+import threading
+import time
+from typing import Callable
+
+import zenoh
+
+# Open session
+session = zenoh.open(zenoh.Config())
+
+
+# Test support: send data in background
+def send_data():
+    time.sleep(1)
+    for i in range(5):
+        session.put("key/expression", f"sample_{i}")
+
+
+threading.Thread(target=send_data, daemon=True).start()
+
+
+# [custom_channel]
+class CustomChannel:
+    def __init__(self, max_size=100):
+        self.samples = []
+        self.max_size = max_size
+        self.lock = threading.Lock()
```

Comment:
Not really necessary to have a dedicated lock here, just use `with self.condition` everywhere, it will be more consistent.